packages/seal-elasticsearch-adapter/src/ElasticsearchSearcher.php

```diff
@@ -113,6 +113,10 @@ public function search(Search $search): Result
             ];
         }
 
+        if (null !== $search->distinct) {
+            $body['collapse']['field'] = $search->distinct;
```

Comment:
If its a text field we need the `.raw` field in opensearch and elasticsearch, the getFilterField helper in the specific searcher handles that:

```suggestion
            $body['collapse']['field'] = $this->getFilterField($this->search->index, $this->search->distinct);
```